website/project/views/register.py

```diff
@@ -49,6 +54,151 @@ def node_register_page(auth, **kwargs):
     ret.update(_view_project(node, auth, primary=True))
     return ret
 
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_be_public_registration
+def node_registration_retraction_get(auth, **kwargs):
+    """Prepares node object for registration retraction page.
+
+    :return: serialized Node to be retracted
+    :raises: 400: BAD_REQUEST if registration already pending retraction
+    """
+
+    node = kwargs['node'] or kwargs['project']
+    if node.retraction.pending_retraction:
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': 'Invalid Request',
+            'message_long': 'This registration is already pending a retraction.'
+        })
+
+    return serialize_node(node, auth, primary=True)
+
+@must_be_valid_project
+@must_have_permission(ADMIN)
+@must_be_public_registration
+def node_registration_retraction_post(auth, **kwargs):
+    """Handles retraction of public registrations
+
+    :param auth: Authentication object for User
+    :return: Redirect URL for successful POST
+    """
+
+    node = kwargs['node'] or kwargs['project']
+    data = request.get_json()
+
+    try:
+        node.retract_registration(auth.user, data['justification'])
+        node.save()
+    except ValidationValueError:
+        raise HTTPError(http.BAD_REQUEST)
+
+    # Email project admins
+    admins = [contrib for contrib in node.contributors if node.has_permission(contrib, 'admin')]
+    for admin in admins:
+            _send_retraction_email(
+                node,
+                admin,
+                node.retraction.approval_state[admin._id]['approval_token'],
+                node.retraction.approval_state[admin._id]['disapproval_token'],
+            )
+
+    return {'redirectUrl': node.web_url_for('view_project')}
+
+def _send_retraction_email(node, user, approval_token, disapproval_token):
```

Comment:
I originally had this in the model... I'm not sure why but I pulled it into the view as it seemed like a more appropriate place to put it.